db/census_membership.go

```diff
@@ -154,9 +154,8 @@ func (ms *MongoStorage) SetBulkCensusMembership(
 		return nil, fmt.Errorf("failed to get published census: %w", err)
 	}
 
-	org := ms.organizations.FindOne(ctx, bson.M{"_id": census.OrgAddress})
-	if org.Err() != nil {
-		return nil, ErrNotFound
+	if _, _, err := ms.Organization(census.OrgAddress, false); err != nil {
```

Comment:
This will throw because of the double lock that we discussed earlier

Reply:
Use the `ms.organization` instead

Reply:
but there is no lock before this line, in fact we call `ms.Census` just above

Reply:
oooops :-|